### PR TITLE
Added release workflow

### LINKED ISSUE #55: Automate build / release
- [ ] Use github for building
- [ ] Auto increment version
- [ ] Auto-update changelog
- [ ] Publish to vscodium marketplace
- [ ] Publish to vscode marketplace
- [ ] Disable user push to master, except from PR

### PR DESCRIPTION
**Pull request by bard83**
Enabled build, automatic CHANGELOG update and publish artifact.
Close #55.

### CODE REVIEW

**Comment by bard83** [comment]
In some cherry picked commit, have been introduced changes in some class. It should be checked whether it has been introduced some regression.

**Comment by KreativJos** [comment]
All seems correct to me.
The only thing I'd like to change is the correct way of using semantic versioning:

- Major version increment if something breaks (should be done by hand)
- Minor increment if something is added
- Patch increment if something is fixed

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        name: ${{ env.EXTENSION_OUTPUT_FILE }}
+        path: ${{ env.EXTENSION_OUTPUT_FILE }}
+        retention-days: 30
+    - name: Publish Minor
+      run: yarn publish --minor
+      env:
+        VSCE_PAT: ${{ secrets.VSCE_PAT }}
+      if: "contains(github.event.release.body, 'Fixed')"
+    - name: Publish Major
+      run: yarn publish --major
```
Comment (KreativJos) [inline .github/workflows/release.yml:70]: So simply use minor here?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      run: yarn package -o ${{ env.EXTENSION_OUTPUT_FILE }}
+
+    - name: Upload artifact
+      uses: actions/upload-artifact@v2
+      with:
+        name: ${{ env.EXTENSION_OUTPUT_FILE }}
+        path: ${{ env.EXTENSION_OUTPUT_FILE }}
+        retention-days: 30
+    - name: Publish Minor
+      run: yarn publish --minor
```
Comment (KreativJos) [inline .github/workflows/release.yml:65]: And use patch here?

**Comment by bard83** [comment]
All changes addressed

**Review by KreativJos — APPROVED** [review]

**Comment by KreativJos** [comment]
Hmm, I'll have to figure out why the script doesn't work in the release:
https://github.com/KreativJos/csharpextensions/runs/6064055532?check_suite_focus=true

**Comment by bard83** [comment]
Probably the check out refers to branch main and not to master

**Pull request merged**